plot/plot.app.js

```diff
@@ -684,13 +690,57 @@
+                                                                //     Title: traceLabel,
+                                                                // };
+                                                                try{
+                                                                    traceLabel=ERMrest.renderMarkdown(ERMrest.renderHandlebarsTemplate(trace.legend_markdown_pattern,$rootScope.templateParams),true);
+                                                                    //Defined regex pattern to extract the url in markdown pattern
+                                                                    var matchs =/<a\b.+href="([^\n\r]*)"\s/ig.exec(traceLabel);
+                                                                    values.legend_markdown_pattern.push(matchs[1]+qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid);
+                                                                }catch(error){
+                                                                    console.log(error)
+                                                                }
+                                                                delete $rootScope.templateParams;
+                                                            }
+                                                            values.labels.push(traceLabel);
+                                                            //Added the legend column names to text variable so that the tooltip name does not contain a link if the legends contains a link
+                                                            values.text.push(row[trace.legend_col]);
                                                         }
                                                         values.values.push(formatData(row[trace.data_col], plot.config ? plot.config.format_data : false, "pie"));
+                                                    
                                                     });
                                                     plot_values.data.push(values);
+                                                    if(plot.config.hasOwnProperty("title_markdown_pattern")){
+                                                        var link=/\((.*?)\)/ig.exec(plot.config.title_markdown_pattern)[1];
+                                                        var qCharacter = link.indexOf("?") !== -1 ? "&" : "?";
+                                                        $rootScope.templateParams= {
+                                                            Title: layout.title,
+                                                        };
+                                                        //Appending pcid and ppid to the URL request
+                                                        plot.config.title_markdown_pattern.replace(/\((.*?)\)/ig.exec(plot.config.title_markdown_pattern)[1],link+qCharacter+"pcid="+contextUrlParams.cid+"&ppid="+contextUrlParams.pid)
```

Comment:
don't change the value that is in the configuration file. copy `plot.config.title_markdown_pattern` to a local variable and modify that.

Reply:
Made changes